taca/utils/bioinfo_tab.py

```diff
@@ -0,0 +1,163 @@
+                    update_statusdb(run_dir)
+
+
+def update_statusdb(run_dir):
+    project_ids=get_ss_projects(run_dir)
+    run_name = os.path.basename(os.path.abspath(run_dir))
+    status=get_status(run_dir)
+    couch=setupServer(CONFIG)
+    db=couch['bioinfo_analysis']
+    view = db.view('full_doc/pj_run_to_doc')
+    for p in project_ids:
+        obj={'run_id':run_name, 'project_id':p, 'status':status}
+        if len(view[[p, run_name]].rows) == 1:
+            remote_doc= view[[p, run_name]].rows[0].value
+            remote_status=remote_doc["status"]
+            if remote_status in ['Incoming', 'Sequencing Done', 'Demultiplexing', 'Demultiplexed', 'Transferring']:
+                final_obj=merge(obj, remote_doc)
+                logger.info("saving {} {} as  {}".format(run_name, p, status))
+                db.save(final_obj)
+        else:
+            logger.info("saving {} {} as  {}".format(run_name, p, status))
+            db.save(obj)
+
+
+
+def get_status(run_dir):
+    status='Incoming'
+
+    run_name = os.path.basename(os.path.abspath(run_dir))
+    xten_dmux_folder=os.path.join(run_dir, 'Demultiplexing')
+    xten_dmux_stats=os.path.join(xten_dmux_folder, 'Stats', 'DemultiplexingStats.xml')
+    unaligned_folder=glob.glob(os.path.join(run_dir, 'Unaligned_*'))
+    unaligned_dmux_stats=glob.glob(os.path.join(run_dir, 'Unaligned_*', 'Basecall_Stats_*', 'Demultiplexing_Stats.htm'))
+    taca_transfer=os.path.join(CONFIG['analysis']['status_dir'], 'transfer.tsv')
+    old_transfer=CONFIG['bioinfo_tab']['b5_transfer']
+
+    
+
+
+    if os.path.exists(os.path.join(run_dir, 'RTAComplete.txt')):
+        status='Sequencing Done'
+    if os.path.exists(xten_dmux_folder) or unaligned_folder:
+        status="Demultiplexing"
+    if os.path.exists(xten_dmux_stats) or unaligned_dmux_stats:
+        status='Demultiplexed'
+    if os.path.exists(os.path.join(run_dir, 'transferring')):
+        status='Transferring'
+    if os.path.exists(taca_transfer):
+        with open(taca_transfer) as t_file:
+            for line in t_file:
+                if run_name in line:
+                    status='Ongoing'
+
+    if os.path.exists(old_transfer):
+        with open(old_transfer) as t_file:
+            for line in t_file:
+                if run_name in line:
+                    elements=line.split("\s")
+                    if len(elements)==2:
+                        status='Transferring'
+                    else:
+                        status='Ongoing'
+
+    return status
+
+
+
+
+def get_ss_projects(run_dir):
+    pattern=re.compile("(P[0-9]{3,5})_[0-9]{3,5}")
+    project_ids=set()
+    run_name = os.path.basename(os.path.abspath(run_dir))
+    current_year = '20' + run_name[0:2]
+    run_name_components = run_name.split("_")
+    FCID = run_name_components[3][1:]
+
+    xten_samplesheets_dir = os.path.join(CONFIG['bioinfo_tab']['xten_samplesheets'],
```

Comment:
same answer as before : the only comment I heard about this PR was "it does not work", so I'm not really keen on using it.